airflow/models.py

```diff
@@ -1750,14 +1750,22 @@ def get_template_context(self, session=None):
         if 'tables' in task.params:
             tables = task.params['tables']
 
-        ds = self.execution_date.isoformat()[:10]
+        ds = self.execution_date.strftime('%Y-%m-%d')
         ts = self.execution_date.isoformat()
-        yesterday_ds = (self.execution_date - timedelta(1)).isoformat()[:10]
-        tomorrow_ds = (self.execution_date + timedelta(1)).isoformat()[:10]
+        yesterday_ds = (self.execution_date - timedelta(1)).strftime('%Y-%m-%d')
+        tomorrow_ds = (self.execution_date + timedelta(1)).strftime('%Y-%m-%d')
 
         prev_execution_date = task.dag.previous_schedule(self.execution_date)
         next_execution_date = task.dag.following_schedule(self.execution_date)
 
+        next_ds = None
```

Comment:
per discussion, I wonder if we could have a better default here .

Reply:
The idea of {{ next_ds }} is just {{ next_execution_date }} in `YYYY-MM-DD` format so that we don't need to write something like {{ ds_format(next_execution_date, "%Y-%m-%d")}}

Thus I think we should make sure that  the behavior of {{ next_ds }} aligns with 
{{ next_execution_date }}.

So I think we should do either:
option 1:
If {{ next_execution_date }} is None, we set {{ next_ds }} to None.

option 2:
We make sure {{ next_execution_date }} is never None. But this will change the fact that 
{{ next_execution_date }} is `None` when `schedule_interval="@once".

Reply:
cool, Default with none is ok in this case.